src/proxy.rs

```diff
@@ -190,6 +187,37 @@ impl LocalWorkloadInformation {
     pub fn workload_info(&self) -> Arc<WorkloadInfo> {
         self.wi.clone()
     }
+
+    pub fn as_fetcher(&self) -> Arc<LocalWorkloadFetcher> {
+        LocalWorkloadFetcher::new(self.wi.clone(), self.state.clone())
+    }
+}
+
+/// LocalWorkloadFetcher is essentially LocalWorkloadInformation without CA access.
```

Comment:
Eventually it might make sense to invert this a bit and do

```
pub struct LocalWorkloadInformation {
    fetcher: LocalWorkloadFetcher
    // full_cert_manager gives access to the full SecretManager. This MUST only be given restricted
    // access to the appropriate certificates
    full_cert_manager: Arc<SecretManager>,
}
```

but this makes sense as well.